src/config.c

```diff
@@ -128,3 +171,109 @@ void config_dump()
 
   printf("}}\n");
 }
+
+/*
+  parse array reference xxx[idx1][idx2]
+  if an array referene is found then name is truncated at the first [
+*/
+static void parse_array_name(char *name, int8_t *idx1, int8_t *idx2)
+{
+    char *p = strchr(name,'[');
+    if (p == NULL) {
+        return;
+    }
+    *p = 0;
+    int8_t i = strtol(p+1, &p, 10);
+    if (!p || *p != ']') {
+        return;
+    }
+    *idx1 = i;
+    if (p[1] == '[') {
+        i = strtol(p+2, &p, 10);
+        if (!p || *p != ']') {
+            return;
+        }
+        *idx2 = i;
+    }
+}
+
+
+/*
+  show a config variable
+ */
+void config_show(char *name)
+{
+    int8_t idx1=-1, idx2=-1;
+    parse_array_name(name, &idx1, &idx2);
+    for (uint8_t i=0; i<CONFIG_TABLE_LEN; i++) {
+        const struct ctable *c = &config_table[i];
+        if (strcmp(name, c->name) == 0) {
+            switch (c->type) {
+            case CTYPE_UINT8:
+                printf("%s=%u\n", name, *(uint8_t *)c->ptr);
+                break;
+            case CTYPE_UINT16:
+                printf("%s=%u\n", name, *(uint16_t *)c->ptr);
+                break;
+            case CTYPE_INT16:
+                printf("%s=%d\n", name, *(int16_t *)c->ptr);
+                break;
+            case CTYPE_INT16_ARRAY:
+                if (idx1 == -1) {
+                    for (uint8_t i=0; i<c->tsize1; i++) {
+                        printf("%s[%d]=%d\n", name, i, ((int16_t *)c->ptr)[i]);
+                    }
+                } else if (idx1 >= 0 && idx1 < c->tsize1) {
+                    printf("%s[%d]=%d\n", name, idx1, ((int16_t *)c->ptr)[idx1]);
+                }
+                break;
+            case CTYPE_UINT16_2D_ARRAY:
+                if (idx1 >= 0 && idx1 < c->tsize1 && idx2 == -1) {
```

Comment:
Consolidation of the idx1 constraints with those in the else clause would make things a little clearer.